docs/src/main/sphinx/security/oauth2.rst

```diff
@@ -0,0 +1,126 @@
+    http-server.authentication.oauth2.token-url=https://authorization-server.com/token
+    http-server.authentication.oauth2.jwks-url=https://authorization-server.com/.well-known/jwks.json
+    http-server.authentication.oauth2.client-id=CLIENT_ID
+    http-server.authentication.oauth2.client-secret=CLIENT_SECRET
+
+In order to enable OAuth 2.0 authentication for the Web UI, the following
+properties need to be added as well:
+
+.. code-block:: text
+
+    web-ui.enabled=true
+    web-ui.authentication.type=oauth2
+
+.. list-table:: Properties
+   :widths: 40 60
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - ``http-server.authentication.type``
+     - Enable OAuth 2.0 authentication for the Trino coordinator. Must  be set
+       to ``oauth2``.
+   * - ``http-server.https.enabled``
+     - Enables HTTPS access for the Trino coordinator. Should be set to
+       ``true``. Default value is ``false``.
+   * - ``http-server.https.port``
+     - HTTPS server port.
+   * - ``http-server.https.keystore.path``
+     - The location of the Java Keystore file that will be used to secure TLS.
+   * - ``http-server.https.keystore.key``
+     - The password for the keystore. This must match the password you specified
+       when creating the keystore.
+   * - ``http-server.process-forwarded``
+     - Enable treating forwarded HTTPS requests over HTTP as secure.  Requires
+       the ``X-Forwarded-Proto`` header to be set to ``https`` on forwarded
+       requests. Default value is ``false``.
+   * - ``http-server.authentication.oauth2.auth-url``
+     - The authorization URL. The URL a user agent will be redirected to in
+       order to begin the OAuth 2.0 authorization process.
+   * - ``http-server.authentication.oauth2.token-url``
+     - The token URL. The URL of the endpoint which will be used by Trino in
+       order to obtain an access token.
+   * - ``http-server.authentication.oauth2.jwks-url``
+     - The JSON Web Key Set (JWKS) URL. JWKS is a set of keys containing the
+       public key used to verify any JSON Web Token (JWT) issued by the
+       authorization server.
+   * - ``http-server.authentication.oauth2.client-id``
+     - The public identifier of the Trino client.
+   * - ``http-server.authentication.oauth2.client-secret``
+     - The secret used to authorize Trino client with the authorization server.
+   * - ``http-server.authentication.oauth2.audience``
+     - The audience of a JSON Web Token. This value will be used as the target
+       audience of an access token requested by the Trino coordinator as well as
+       the required audience of an access token included in user requests to
+       the coordinator.
+   * - ``http-server.authentication.oauth2.scopes``
+     - Scopes requested by the server during the authorization challenge.
+   * - ``http-server.authentication.oauth2.challenge-timeout``
+     - Maximum duration of the authorization challenge.
```

Comment:
what unit? maybe some example values? Is there a default value or a suggested value?